project-management/src/main/java/life/qbic/projectmanagement/domain/model/measurement/ProteomicsMeasurement.java

```diff
@@ -162,51 +158,43 @@ private static void evaluateMandatoryMetadata(ProteomicsMethodMetadata method)
    * entity with many describing properties about provenance and instrumentation.
    *
    * @param projectId
-   * @param sampleIds the sample ids of the samples the measurement was performed on. If more than
-   *                  one sample id is provided, the measurement is considered to be performed on a
-   *                  pooled sample
    * @return
    * @throws IllegalArgumentException in case there are missing required metadata.
    * @since 1.0.0
    */
-  public static ProteomicsMeasurement create(ProjectId projectId, Collection<SampleId> sampleIds,
-      MeasurementCode measurementCode, Organisation organisation, ProteomicsMethodMetadata method)
+  public static ProteomicsMeasurement create(ProjectId projectId,
+      MeasurementCode measurementCode, Organisation organisation, ProteomicsMethodMetadata method,
+      Collection<ProteomicsSpecificMeasurementMetadata> proteomicsSpecificMeasurementMetadata)
       throws IllegalArgumentException {
-    if (sampleIds.isEmpty()) {
-      throw new IllegalArgumentException(
-          "No sample ids provided. At least one sample id must provided for a measurement.");
-    }
     requireNonNull(method.instrument());
     requireNonNull(measurementCode);
+    requireNonNull(proteomicsSpecificMeasurementMetadata);
```

Comment:
```suggestion
    requireNonNull(proteomicsSpecificMeasurementMetadata, "proteomicsSpecificMeasurementMetadata must not be null");
```
As Java introduced helpful NullPointer messages this is NIT!
This is low effort and can help in debugging.

Reply:
any references for reading up?

Reply:
Sure:
The require non null method accepts a helpful message. Prior to Java 14, there was no information on what was null but only a generic nullpointer message. 

https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-java.lang.String-
https://openjdk.org/jeps/358